File: LeetCode/1491/solution.js
/**
 * @param {number[]} salary
 * @return {number}
 */
var average = function(salary) {
    const highest = Math.max(...salary);
    const lowest = Math.min(...salary);

    const total = salary.reduce((a, c) => a + c, 0);
    const reduced = total - highest - lowest;

    const average = reduced / (salary.length - 2);

    return average;
};

console.log(average([1000,2000,3000]))

File: LeetCode/1491/solution.py
from typing import List

def average(salary: List[int]) -> float:
        highest = max(salary)
        lowest = min(salary)
        n = len(salary)

        total = sum(salary)
        reduced = total - highest - lowest

        average = reduced / (n - 2)

        return average

result = average(salary=[1000,2000,3000])
print(result)